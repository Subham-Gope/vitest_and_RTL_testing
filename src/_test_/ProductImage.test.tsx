// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ProductImage from "../components/ProductImage";

// describe("Testing the ProductImage.tsx file", () => {
//   test("displays the preview image on hovering the thumbnail", async () => {
//     // Setting up the userEvent
//     const user = userEvent.setup();

//     // Setting up the debugger
//     screen.debug();

//     // setting up the snapshot matching
//     const { container } = render(<ProductImage />);
//     expect(container).toMatchSnapshot();

//     const thumbnail = screen.getByAltText(/thumbnail/i);
//     expect(thumbnail).toBeInTheDocument();

//     // Initiating the user interaction
//     await user.hover(thumbnail);
//     const preview = screen.queryByAltText(/preview/i);
//     expect(preview).toBeInTheDocument();

//     await user.unhover(thumbnail);
//     expect(preview).not.toBeInTheDocument();
//   });
// });
