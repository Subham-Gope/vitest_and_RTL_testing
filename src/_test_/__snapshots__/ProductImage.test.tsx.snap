// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ProductImage.tsx file > displays the preview image on hovering the thumbnail 1`] = `
<div>
  <div>
    <img
      alt="thumbnail"
      src="https://images.pexels.com/photos/3270223/pexels-photo-3270223.jpeg"
    />
  </div>
</div>
`;
